plugins/main/host-device/host-device.go

```diff
@@ -230,117 +230,93 @@ func cmdDel(args *skel.CmdArgs) error {
+
+	tempNS, err := ns.TempNetNS()
 	if err != nil {
-		return nil, fmt.Errorf("failed to get host namespace: %v", err)
+		return nil, fmt.Errorf("failed to create temp NetNS: %v", err)
 	}
+	defer tempNS.Close()
 
-	// Devices can be renamed only when down
-	if err = netlink.LinkSetDown(hostDev); err != nil {
-		return nil, fmt.Errorf("failed to set %q down: %v", hostDev.Attrs().Name, err)
+	if err = netlink.LinkSetNsFd(hostDev, int(tempNS.Fd())); err != nil {
+		return nil, fmt.Errorf("failed to move %q to temp NS: %v", hostDev.Attrs().Name, err)
 	}
+	// detroying tempNS will move the interface back to the host NS
 
-	// restore original link state in case of error
-	defer func() {
+	if err = tempNS.Do(func(_ ns.NetNS) error {
+		tempNsDev, err := netlink.LinkByName(hostDevName)
 		if err != nil {
-			if origLinkFlags&net.FlagUp == net.FlagUp && hostDev != nil {
-				_ = netlink.LinkSetUp(hostDev)
-			}
-		}
-	}()
-
-	hostDev, err = setTempName(hostDev)
-	if err != nil {
-		return nil, fmt.Errorf("failed to rename device %q to temporary name: %v", hostDevName, err)
-	}
-
-	// restore original netdev name in case of error
-	defer func() {
-		if err != nil && hostDev != nil {
-			_ = netlink.LinkSetName(hostDev, hostDevName)
+			return fmt.Errorf("failed to find %q in temp NS: %v", hostDevName, err)
 		}
-	}()
 
-	if err = netlink.LinkSetNsFd(hostDev, int(containerNs.Fd())); err != nil {
-		return nil, fmt.Errorf("failed to move %q to container ns: %v", hostDev.Attrs().Name, err)
-	}
-
-	var contDev netlink.Link
-	tempDevName := hostDev.Attrs().Name
-	if err = containerNs.Do(func(_ ns.NetNS) error {
-		var err error
-		contDev, err = netlink.LinkByName(tempDevName)
-		if err != nil {
-			return fmt.Errorf("failed to find %q: %v", tempDevName, err)
+		// Live device rename is only supported since kernel 6.2
+		// https://github.com/torvalds/linux/commit/bd039b5ea2a91ea707ee8539df26456bd5be80af
+		if err = netlink.LinkSetDown(tempNsDev); err != nil {
+			return fmt.Errorf("failed to set %q down: %v", tempNsDev.Attrs().Name, err)
 		}
 
-		// move netdev back to host namespace in case of error
+		// restore original link state in case of error
 		defer func() {
 			if err != nil {
-				_ = netlink.LinkSetNsFd(contDev, int(defaultNs.Fd()))
-				// we need to get updated link object as link was moved back to host namepsace
-				_ = defaultNs.Do(func(_ ns.NetNS) error {
-					hostDev, _ = netlink.LinkByName(tempDevName)
-					return nil
-				})
+				origLinkFlags := hostDev.Attrs().Flags
+				if origLinkFlags&net.FlagUp == net.FlagUp && tempNsDev != nil {
+					_ = netlink.LinkSetUp(tempNsDev)
+				}
 			}
 		}()
 
-		// Save host device name into the container device's alias property
-		if err = netlink.LinkSetAlias(contDev, hostDevName); err != nil {
-			return fmt.Errorf("failed to set alias to %q: %v", tempDevName, err)
-		}
 		// Rename container device to respect args.IfName
-		if err = netlink.LinkSetName(contDev, ifName); err != nil {
```

Comment:
I can't find where you set the device to `contDev` now, that should be the final name

Reply:
`contDev` is `netlink.Link`, the interface name is `ifName`, see new line 270

Reply:
reworked a bit more the PR with way more comments, hope it helps